### PR TITLE
Added more FMA functions with tests

### PR DESCRIPTION
**Pull request by JishinMaster**
This pull request is the last split of the original #83.

I have added fmsub_ps, fmnadd_ps, fmnsub_ps, and the associated tests.
I had to lower the input vector in the test functions and use an epsilon of 0.0001f because those functions are less precise than just using mul and add, and they seem to diverge with big float numbers (I use qemu, it might be worth checking on real hardware).

### CODE REVIEW

**Comment by jserv** [comment]
Non-SSE intrinsics are considered as the additional work to SSE2NEON as #82 stated.

File: sse2neon.h [hunk 1]
```diff
@@ -1307,7 +1307,7 @@ FORCE_INLINE __m128i _mm_shuffle_epi_3332(__m128i a)
 
 // Shuffle packed 8-bit integers in a according to shuffle control mask in the
 // corresponding 8-bit element of b, and store the results in dst.
-// https://software.intel.com/sites/landingpage/IntrinsicsGuide/#text=_mm_shuffle_epi8
+// https://software.intel.com/sites/landingpage/IntrinsicsGuide/#text=_mm_shuffle_epi8&expand=5146
```
Comment (marktwtn) [inline sse2neon.h:1310]: Revert it.

File: sse2neon.h [hunk 2]
```diff
@@ -2003,7 +2003,7 @@ FORCE_INLINE int _mm_movemask_ps(__m128 a)
 
 // Compute the bitwise AND of 128 bits (representing integer data) in a and
 // mask, and return 1 if the result is zero, otherwise return 0.
-// https://software.intel.com/sites/landingpage/IntrinsicsGuide/#text=_mm_test_all_zeros
+// https://software.intel.com/sites/landingpage/IntrinsicsGuide/#text=_mm_test_all_zeros&expand=5871
```
Comment (marktwtn) [inline sse2neon.h:2006]: Ditto.

File: sse2neon.h [hunk 3]
```diff
@@ -2510,11 +2510,46 @@ FORCE_INLINE __m128 _mm_fmadd_ps(__m128 a, __m128 b, __m128 c)
 #endif
 }
 
+// Computes the fused multiple add product of 32-bit floating point numbers.
+//
+// Return Value
+// Multiplies A and B, and adds C to the temporary result before returning it.
```
Comment (marktwtn) [inline sse2neon.h:2516]: `adds` -> `subtracts`

Comment (marktwtn) [inline sse2neon.h:2516]: Shall we unify these descriptions of `FMA` functions with the link of `Intel Intrinsics Guide`?
For example:
```
Multiply packed single-precision (32-bit) floating-point elements in a and b,
subtract packed elements in c from the intermediate result, and store the results in dst.
```

**Comment by jserv** [comment]
Header `<immintrin.h>` is used for AVX, AVX2 and FMA instructions. We need to figure out the way to manipulate. Non-SSE intrinsics are considered as the additional work to SSE2NEON as #82 stated.

**Comment by JishinMaster** [comment]
Can we use something like "ifdef USE_FMA"‚  and enable only for those who want it?
Or use a completely different "sse2neon_optional.h"  file ? Doing so we could also implement some MMX instructions from older programs which have not migrated to SSE2.

**Comment by jserv** [comment]
Let's close this pull request in favor of #82.

**Pull request closed without merging**